detection-rules/attachment_html_excessive_const_declarations.yml

```
@@ -0,0 +1,42 @@
+name: "Attachment: HTML file with excessive 'const' declarations and abnormally long timeouts"
+description: "Detects messages with HTML attachments containing multiple 'const' declarations while excluding legitimate Gmail messages. This is evidence of potential code injection or obfuscation techniques."
+type: "rule"
+severity: "high"
+source: |
+  type.inbound
+  and any(attachments,
+          (
+            .file_extension in~ ("html", "htm", "shtml", "dhtml")
+            or .file_type == "html"
+          )
+          and strings.count(file.parse_html(.).raw, 'const') >= 7
+          and not regex.contains(file.parse_html(.).raw,
+                                 "<!-- saved from url=.{0,7}https://mail.google.com/mail/u/0/#inbox/"
+          )
+          and length(file.parse_html(.).raw) < 50000
+          // long timeouts
+          and regex.icontains(file.parse_html(.).raw,
+                              'setTimeout\(\s*function\s*\(.*?\)\s*\{[\s\S]*?\},\s*\d+\);'
```

```suggestion
                              'setTimeout\(\s*(?:function\s*)?\(.*?\)\s*(?:=>\s*)?{[\s\S]*?\},\s*\d+\);'
```
catch another form of this
17f72d467641964f1d0126ec374b485748130dd76a19a6334ce5eccfa55d67b4